src/robot/libraries/BuiltIn.py

```diff
@@ -876,31 +970,51 @@ def get_count(self, item1, item2):
         self.log('Item found from the first item %d time%s' % (count, s(count)))
         return count
 
-    def should_not_match(self, string, pattern, msg=None, values=True):
+    def should_not_match(self, string, pattern, msg=None, values=True, ignore_case=False):
         """Fails if the given ``string`` matches the given ``pattern``.
 
         Pattern matching is similar as matching files in a shell, and it is
         always case-sensitive. In the pattern ``*`` matches to anything and
         ``?`` matches to any single character.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``first`` and ``second`` should be compared case-insensitively,
+        provided that ``first`` and ``second`` are string types.  See `Boolean
```

Comment:
Similarly as with other keywords that should *always* be used with strings, `provided that ... are string types` is unnecessary.